Remove deprecated syntax support

aeslaughter opened the pull request:
This updates all markdown in MOOSE to use the latest version of the inline command syntax and removes the deprecated syntax.

I am expecting this to fail application documentation, but I am working on updating those now.

This also updates the acronym extension testing to me unittest based.

(see #13611)

<!--
INCLUDE THE FOLLOWING IN THE PR DESCRIPTION
- Explain relevant design information for your change.
- Follow the [Coding Standards](http://mooseframework.org/wiki/CodeStandards/).
- Submit or improve [Test Cases](http://mooseframework.org/wiki/MooseTraining/testing/).
- Reference a specific issue, place "refs #<issue>" or "closes #<issue>" (e.g., #closes #1234).
-->

aeslaughter committed:
Update inline command syntax

(refs #13611)

aeslaughter committed:
Unit test acronym extension

(refs #13611)

aeslaughter committed:
Remove acronym from MooseDocs test application; tests have been moved to unit tests classes

(refs #13611)

aeslaughter committed:
Update gold for removal of acronym pages

(refs #13611)

aeslaughter committed:
Remove deprecated cite syntax

(refs #13611)

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/356512/) on 092bd65 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/13625/site)

This comment will be updated on new commits.

dschwen commented on framework/doc/content/source/base/MooseApp.md:
```diff
@@ -1,4 +1,4 @@
 # MooseApp
 
-The `MooseApp` object is the base class for !!acro MOOSE!! based applications and serves as the
+The `MooseApp` object is the base class for [!ac!MOOSE] based applications and serves as the
```
Why is it `[!ac!MOOSE]` and not `[!ac](MOOSE)` analogous to the cite syntax?

aeslaughter replied:
The main reasons is that I wanted to make the acronym syntax as compact as possible so it is quick to use. There are also some implementation reasons that are not super important. 

If you feel it makes the markdown better I will change it, but can we merge this first? I will then add support for both and print a deprecated warning for the old version.

aeslaughter replied:
I added the new version, we should probably change it. The consistency is better then the one character savings. 

We can merge this with both and then go update the apps to avoid submodule stuff.

aeslaughter committed:
Use content form of acronym command for consistency

(refs #13611)

aeslaughter committed:
Update acronym syntax

(refs #13611)

dschwen reviewed (dismissed):
Looks good to me (only reviewed all non md files, and did a spot check on the md files). Thanks for the acronym syntax change!

aeslaughter commented:
This will still be a bit, there are some apps that need to be updated but all the PRs are up.

aeslaughter committed:
Update inline links to avoid interaction with inline command

It was reported that the following string wasn't parsing correctly in MooseDocs, it was tracked down to be the LinkInline class in core.py that was capturing up to the final ).

like in [fig:yieldsurface] (after [!citet](chiang1994anew)).

To fix it, the text within the [] for links is not allowed to contain additional [, which seems reasonable. Another way to fix it
is just to begin a new line with the '(after...'

(refs #13611)

aeslaughter committed:
Update additional syntax

(refs #13611)

aeslaughter committed:
Remove [] from link syntax

(refs #13611)

aeslaughter committed:
The LinkInline fix doesn't work

(refs #13611)

aeslaughter committed:
Limit shortcuts to exclude spaces

(refs #13611)
